spec/apollo-server-spec.yaml

```diff
@@ -0,0 +1,56 @@
+swagger: "2.0"
+info:
+  description: "This is the backend component API exposing a GraphQL server. The [GraphQL spec](https://spec.graphql.org/June2018/) dives deep into the philosophy of GraphQL. This spec models and showcases the request/response formats expected from the backend service."
+  version: "1.0.0"
+  title: "Road Rules Search Engine"
+tags:
+  - name: "Backend service"
+schemes:
+  - "http"
+paths:
+  /:
+    post:
+      tags:
+        - "Backend service"
+      summary: "GraphQL endpoint"
+      description: "Base path handles any incoming request and re-routes it to the requested resolver."
+      consumes:
+        - "application/json"
+      produces:
+        - "application/json"
+      parameters:
+        - in: "body"
+          name: "body"
+          description: "Any GraphQL request ( query/mutation/subscription )"
+          schema:
+            $ref: '#/definitions/Request'
+      responses:
+        "200":
+          description: "Default response format"
+          schema:
+            $ref: '#/definitions/Response'
```

Comment:
will a 200 response always contain an empty `errors` array?

Reply:
Nope, only if there are errors available.

Reply:
Okay, so by what I understand from the YAML is that when an operation is successful it will always return a 200 and this (even when errors happen)
`{
  "data": {},
  "errors": [
    {}
  ]
}`

You can add a response other than a 200 that signals an error and you can return the error message there.

Reply:
No, the GraphQL spec specifically mentions that you always get a 200 response status.

The Apollo Server we're using always returns this code.